### DISCUSSION TITLE
Is partial component update possible?

### DISCUSSION THREAD

**Discussion by Alex-Sichkar** [tetra-framework/tetra#20] [Q&A]
Hi @samwillis,

What for? For example, we have a dropdown with live search functionality that is positioned using javascript adding certain styles. During the search, the entire component is redrawn and positioning is lost. We can reposition it, but that will cause flicker. It would be nice to be able to redraw only the contents of the menu. Though my approach may not be good, perhaps I should split the component apart? What do you think? Thank you.

**Comment by nerdoc** [comment]
This would be like `hx-select` in the HTMX world.

**Comment by samwillis** [comment]
@Alex-Sichkar Apologies for slow reply, missed this one.

If you use Alpine to position your dropdown the state should be maintained between successive server re-renders. Other attributes/styles added by outside JS would be lost. This is the same as you would experience with React/Vue and mixing outside JS that interferes with the DOM.

The alternative is to split the component in two, a Dropdown and DropdownContent component, only updating the latter. The concept is that a sub-component is the partial.
